### PR TITLE
Drop Python 2 Support

### LINKED ISSUE #129: Drop Python 2 Support
Python 2 is [deprecated as of 2020](https://www.python.org/downloads/release/python-2718/).  Dropping Python 2 support would simplify virtually every future change, since it invariably causes annoying and tricky-to-fix problems.  Just one example is in my work on #81.  Tests pass on Python 3 and fail on Python 2.

## Todo
- [x] Remove imports `future`, `backports`, `pathlib2`
- [x] Update CI config
- [x] Update dependencies in `setup.py`
- [x] Remove monkey patch `header_encode_patched` 
- [x] Replace `u""` with `""`
- [x] Remove `coding=utf-8` from all files
- [x] Remove code from tests that accounts for newline differences produced by the email library in Python 2 vs. 3
- [x] Recursively search for "python 2"
- [x] Recursively search for "utf.?8"
- [x] Remove "inherit from object" and update pylint directives
- [x] Update Contributing docs
- [ ] Bump major version before release

### PR DESCRIPTION
**Pull request by awdeorio**
Simplify code, removing Python 2 support.

EDIT: also removes Python <3.6 support.  Python 3.4 and 3.5 are deprecated.

Closes #129

### CODE REVIEW

**Comment by awdeorio** [comment]
I'd appreciate a review from @seshrs if you have time :)

File: mailmerge/sendmail_client.py [hunk 1]
```diff
@@ -20,13 +19,9 @@
 )
 
 
-class SendmailClient(object):
+class SendmailClient:
     """Represent a client connection to an SMTP server."""
 
-    # We need to inherit from object for Python 2 compantibility
```
Comment (seshrs) [inline mailmerge/sendmail_client.py:26]: Lol wow, I’m glad we don’t need this any more.

File: tests/test_ratelimit.py [hunk 2]
```diff
@@ -1,26 +1,20 @@
...
 
 Andrew DeOrio <awdeorio@umich.edu>
 """
 import textwrap
 import datetime
-from future.backports import email
-import future.backports.email.parser  # pylint: disable=unused-import
+from pathlib import Path
+import email
+import email.parser
```
Comment (seshrs) [inline tests/test_ratelimit.py:10]: Hmm, do we not need the pylint disabling comment any more?

**Review by seshrs — APPROVED** [review]
Goodbye Python2!

File: tests/test_ratelimit.py (hunk 2 above)
Comment (awdeorio) [reply tests/test_ratelimit.py:10]: Apparently not!  It's possible that this is due to a pylint improvement, however.

**Pull request merged**